services/web/server/src/simcore_service_webserver/groups_api.py

```diff
@@ -182,6 +184,31 @@ async def list_users_in_group(
         return users_list
 
 
+async def auto_add_user_to_groups(app: web.Application, user_id: int) -> None:
+    user_email: str = await get_user_email(app, user_id)
+    # auto add user to the groups with the right rules
+    engine = app[APP_DB_ENGINE_KEY]
+    async with engine.acquire() as conn:
+        # get the groups where there are inclusion rules and see if they apply
+        query = sa.select([groups]).where(groups.c.inclusion_rules != {})
+        possible_gids = set()
+        async for row in conn.execute(query):
+            email_rules: List[str] = row[groups.c.inclusion_rules].get("emails")
+            for suffix in email_rules:
```

Comment:
SUGGESTION:
The idea of the inclusion rules is very nice.
The concept of "rule" is not well defined but for a first version I guess it is ok
I would only suggest to start with regular expressions matches instead of only an ``endswith`` rule since it is more flexible and does not require a lot more code. Specifically, an inclusion_rules is a json that maps columns of the users table to regular expressions  e.g. for "endswith" it would be writing something like  ``{ users.email: itis.swiss$ }``

Consider this example
```python

import re


z43_group_inclusion_rules = {"email": r"@[\bitis\b|\bspeag\b|\bzmt\b]+\.swiss$"}

users_table = [
    {"user": "foo", "email": "foo@itis.swiss"},
    {"user": "bar", "email": "bar@speag.swiss"},
    {"user": "flo", "email": "flo@flo.swiss"},
]

def main():
    for prop, rule_pattern in z43_group_inclusion_rules.items():

        for user_row in users_table:
            value = user_row[prop]
            if re.search(rule_pattern, value):
                print("Z43 rule applies for ", user_row["user"])
            else:
                print("Z43 rule does NOT apply for ", user_row["user"])


if __name__ == "__main__":
    main()
```
and returns

```bash
 ✗ python re-rules.py
Z43 rule applies for  foo
Z43 rule applies for  bar
Z43 rule does NOT apply for  flo
```

Reply:
great idea. will add.